src/datadreamer/image_generation/sdxl_image_generator.py

```diff
@@ -37,22 +37,40 @@ def _init_gen_model(self):
         Returns:
             tuple: The base and refiner models.
         """
-        base = DiffusionPipeline.from_pretrained(
-            "stabilityai/stable-diffusion-xl-base-1.0",
-            torch_dtype=torch.float16,
-            variant="fp16",
-            use_safetensors=True,
-        )
-        base.enable_model_cpu_offload()
-        refiner = DiffusionPipeline.from_pretrained(
-            "stabilityai/stable-diffusion-xl-refiner-1.0",
-            text_encoder_2=base.text_encoder_2,
-            vae=base.vae,
-            torch_dtype=torch.float16,
-            use_safetensors=True,
-            variant="fp16",
-        )
-        refiner.enable_model_cpu_offload()
+        if self.device == "cpu":
+            base = DiffusionPipeline.from_pretrained(
+                "stabilityai/stable-diffusion-xl-base-1.0",
+                # variant="fp16",
+                torch_dtype=torch.float32,
+                use_safetensors=True,
+            )
+            base.to("cpu")
```

Comment:
I think this block should be simplified by adding the if statement here directly. Same for below.

Reply:
Well, in the CUDA branch, not only the `torch_dtype` is different, but there are `variant` arguments specified, and after the models are also initialized, `base.enable_model_cpu_offload()` is called, none of which is used in the CPU branch. Would you still like me to put the if directly inside, @tersekmatija?

Reply:
Agree with the current approach, because of `base.enable_model_cpu_offload()` and `variant`.